Modules/System/WebDAV/src/Helper/WebDAVRequestHelper.Codeunit.al

```diff
@@ -0,0 +1,180 @@
+
+    [NonDebuggable]
+    procedure Propfind(Uri: Text; Recursive: Boolean) WebDAVOperationResponse: Codeunit "WebDAV Operation Response";
+    var
+        WebDAVHttpContent: Codeunit "WebDAV Http Content";
+        RequestMessage: HttpRequestMessage;
+        Headers: HttpHeaders;
+    begin
+        WebDAVHttpContent.FromXML(PreparePropfindRequest());
+        RequestMessage := PrepareRequestMsg('PROPFIND', Uri, WebDAVHttpContent);
+        RequestMessage.GetHeaders(Headers);
+
+        // Recursive
+        if Headers.Contains('Dephth') then
+            Headers.Remove('Depth');
+        if Recursive then
+            Headers.Add('Depth', 'infinity')
+        else
+            Headers.Add('Depth', '1');
+
+        WebDAVOperationResponse := SendRequest(RequestMessage);
+    end;
+
+    local procedure PrepareRequestMsg(HttpRequestType: Enum "Http Request Type"; Uri: Text): HttpRequestMessage
+    begin
+        Exit(PrepareRequestMsg(Format(HttpRequestType), Uri));
+    end;
+
+    local procedure PrepareRequestMsg(HttpRequestType: Text; Uri: Text) RequestMessage: HttpRequestMessage
+    begin
+        RequestMessage.Method(HttpRequestType);
+        RequestMessage.SetRequestUri(Uri);
+    end;
+
+    local procedure PrepareRequestMsg(HttpRequestType: Enum "Http Request Type"; Uri: Text; WebDAVHttpContent: Codeunit "WebDAV Http Content") RequestMessage: HttpRequestMessage
+    begin
+        Exit(PrepareRequestMsg(Format(HttpRequestType), Uri, WebDAVHttpContent));
+    end;
+
+    local procedure PrepareRequestMsg(HttpRequestType: Text; Uri: Text; var WebDAVHttpContent: Codeunit "WebDAV Http Content") RequestMessage: HttpRequestMessage
+    var
+        Headers: HttpHeaders;
+        HttpContent: HttpContent;
+    begin
+        RequestMessage.Method(HttpRequestType);
+        RequestMessage.SetRequestUri(Uri);
+
+        RequestMessage.GetHeaders(Headers);
+        if WebDAVHttpContent.GetContentLength() > 0 then begin
+
+            HttpContent := WebDAVHttpContent.GetContent();
+            HttpContent.GetHeaders(Headers);
+
+            if Headers.Contains('Content-Type') then
+                Headers.Remove('Content-Type');
+
+            if WebDAVHttpContent.GetContentType() <> '' then
+                Headers.Add('Content-Type', WebDAVHttpContent.GetContentType());
+
+            RequestMessage.Content(HttpContent);
+        end;
+    end;
+
+    local procedure SendRequest(HttpRequestMessage: HttpRequestMessage) OperationResponse: Codeunit "WebDAV Operation Response"
+    var
+        HttpResponseMessage: HttpResponseMessage;
+        HttpClient: HttpClient;
+        IsHandled: Boolean;
+    begin
+        OnBeforeSendRequest(HttpRequestMessage, OperationResponse, IsHandled, HttpRequestMessage.Method());
+
+        if not IsHandled then begin
+            Authorization.Authorize(HttpRequestMessage);
+            if not HttpClient.Send(HttpRequestMessage, HttpResponseMessage) then
+                Error(OperationNotSuccessfulErr);
+
```

Comment:
There's a lot of whitespace below this point in this object. I would see if we can trim that a bit.

Reply:
done